src/System.Collections.NonGeneric/tests/Performance/Perf.ArrayList.cs

```diff
@@ -0,0 +1,476 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Collections.Tests
+{
+    public class Perf_ArrayList
+    {
+        private static volatile int _temp;
+
+        private ArrayList CreateArrayListOfInts(int size = 100000)
+        {
+            var arrayListOfInts = new ArrayList(size);
+            for (int element = 0; element < size; element++)
+            {
+                arrayListOfInts.Add(element);
+            }
+
+            return arrayListOfInts;
+        }
+
+        [Benchmark(InnerIterationCount = 2000000)]
+        public void Add()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                var elements = new ArrayList();
+                using (iteration.StartMeasurement())
+                {
+                    for (int element = 0; element < innerIterationCount; element++)
+                    {
+                        elements.Add(element);
```

Comment:
Since Add takes an object rather than being strongly typed, this is going to box each int, which means we'll also be measuring the costs associated with 2,000,000 allocations.  It might be better to just allocate a single object up front, and then add that same object each time rather than adding element.

Reply:
This is fixed now... I wish I thought about it earlier :)